tests/platforms/web/static/test_build__process_wheel.py

```diff
@@ -93,9 +82,267 @@ def test_process_wheel_no_content(build_command, tmp_path):
+    build_command._process_wheel(wheelfile=wheel_filename, inserts=inserts)
+
+    legacy_msgs = [m for m in warnings if "legacy '/static' CSS detected" in m]
+    assert len(legacy_msgs) == 1
+
+
+def test_process_wheel_non_utf8_insert(build_command, tmp_path):
+    """Non-UTF8 deploy insert raises an UnicodeDecodeError."""
+    bad = b"\xff\xfe\xfa"
+    wheel_path = Path(tmp_path) / "dummy-1.2.3-py3-none-any.whl"
+    with zipfile.ZipFile(wheel_path, "w") as zf:
+        zf.writestr("dummy/deploy/inserts/index.html~header", bad)
+
+    inserts = {}
+    with pytest.raises(UnicodeDecodeError):
+        build_command._process_wheel(wheelfile=wheel_path, inserts=inserts)
+
+
+def test_process_wheel_non_utf8_legacy_css(build_command, tmp_path):
+    """Non-UTF8 legacy CSS raises an UnicodeDecodeError."""
+    bad = b"\xff\xfe\xfa"
+    wheel_path = Path(tmp_path) / "dummy-1.2.3-py3-none-any.whl"
+    with zipfile.ZipFile(wheel_path, "w") as zf:
+        zf.writestr("dummy/static/bad.css", bad)
+
+    inserts = {}
+    with pytest.raises(UnicodeDecodeError):
+        build_command._process_wheel(wheelfile=wheel_path, inserts=inserts)
+
+
+def test_handle_legacy_css_warning_once_and_append(
+    build_command, tmp_path, monkeypatch
+):
+    """Legacy CSS warns once and appends to existing contrib key."""
+    wheel_path = Path(tmp_path) / "dummy-1.2.3.whl"
+    with zipfile.ZipFile(wheel_path, "w") as zf:
+        zf.writestr("dummy/static/one.css", "h1 { x:1; }")
+
+    # Count warnings
+    warnings = []
+    monkeypatch.setattr(build_command.console, "warning", warnings.append)
+
+    inserts = {}
+    with zipfile.ZipFile(wheel_path) as zf:
+        # First call - warn
+        warned = build_command._handle_legacy_css(
+            wheel=zf,
+            path=Path("dummy/static/one.css"),
+            filename="dummy/static/one.css",
+            package_key="pkg 1.0",
+            inserts=inserts,
+            legacy_css_warning=False,
+        )
+        # Pre-seed same contrib key to force append on second call
+        target_map = inserts.setdefault("static/css/briefcase.css", {}).setdefault(
+            "css", {}
+        )
+        key = "pkg 1.0 (legacy static CSS: one.css)"
+        target_map[key] = target_map[key] + "\n/*extra*/"
+        # Second call - no additional warning, content appended
+        warned = build_command._handle_legacy_css(
+            wheel=zf,
+            path=Path("dummy/static/one.css"),
+            filename="dummy/static/one.css",
+            package_key="pkg 1.0",
+            inserts=inserts,
+            legacy_css_warning=warned,
+        )
+
+    # One warning only
+    legacy_msgs = [m for m in warnings if "legacy '/static' CSS detected" in m]
+    assert len(legacy_msgs) == 1
+
+    # Content appended
+    out = inserts["static/css/briefcase.css"]["css"][key]
+    assert "h1 { x:1; }" in out and "/*extra*/" in out
+
+
+def test_handle_insert_register_valid_file(build_command, tmp_path):
```

Comment:
As above - this is testing `_handle_insert()`, so it should be in `test_build__handle_insert.py`.